plugins/onboarding-backend/src/service/router.ts

```diff
@@ -0,0 +1,238 @@
+): Promise<express.Router> {
+  const { logger, identity, database, config } = options;
+  const dbClient = await databaseConnection({ database } as {
+    database: PluginDatabaseManager;
+  });
+  dbClient
+    .raw('select 1+1 as result')
+    .then(() => {})
+    .catch((err: any) => {
+      logger.error(err);
+      process.exit(1);
+    });
+  const urls = config?.getOptionalConfigArray('catalog.locations');
+  let baseRepo = '';
+  urls?.forEach(location => {
+    const type = location.getString('type');
+    if (type === 'onboarding-repo-url') {
+      baseRepo = location.getString('target');
+    }
+  });
+
+  const router = Router();
+  router.use(express.json());
+  /**
+   * @public
+   */
+  router.get('/getChecklists', async (req, res) => {
+    try {
+      const user =
+        (identity && (await identity.getIdentity({ request: req }))) ||
+        DEFAULT_USER;
+      const userId = user?.identity?.userEntityRef;
+      let { groups, roles } = req.query;
+      groups = (groups as string) || 'asro';
+      roles = (roles as string) || '';
+      let roleQuery = '';
+      /**
+       * Creating role base query
+       */
+      if (roles?.length) {
+        roles.split(',').forEach((val: string) => {
+          roleQuery += `${
+            roleQuery.length ? ' or' : ''
+          } “forRoles” Ilike '%${val}%'`;
+        });
+      }
+      let newChecklist: any = {};
+      const prevData = await dbClient
+        .select()
+        .from('checklistResponse')
+        .where({ user: userId });
+      const checklists = await dbClient('checklists')
+        .whereIn('group', groups.split(','))
+        .whereRaw(roleQuery);
+      let showModal = true;
+      const statusArr = prevData.map((item: any) => item.isDone);
+      if (statusArr?.length) showModal = statusArr.some((val: any) => !!!val);
+      if (!checklists?.length) showModal = false;
+      if (checklists?.length) {
+        checklists.forEach((value: any) => {
+          const obj = { ...newChecklist };
+          obj[value.group] = {};
+          if (newChecklist.hasOwnProperty(value.group)) {
+            obj[value.group] = { ...newChecklist[value.group] };
+          }
+          const prevItem = prevData.find(
+            (it: any) => it.checklist_uid === value.checklist_uid,
+          );
+          const formData = JSON.parse(value.formSchema);
+          obj[value.group][value.title] = formData.map((val: any) => {
+            val.checklist_uid = value?.checklist_uid;
+            val.isDone =
+              (prevItem && JSON.parse(prevItem.userResponse)[val.id]) || false;
+            val.checklistHash = value?.checklistHash;
+            return val;
+          });
+          newChecklist = { ...obj };
+        });
+      }
+      res.json({ data: newChecklist, showModal });
+    } catch (error) {
+      res.status(500).send('Internal server error!');
+    }
+  });
+  /**
+   * @public
+   */
+  router.put('/updateStatus', async (req, res) => {
```

Comment:
think we should keep this lowercase and avoid case sensitive.

either `PUT /status` or `PUT /update-status` or something